Runtime/Attributes.cs

```diff
@@ -149,7 +153,91 @@ public IncludeTagsAttribute(params string[] tags)
             Tags = tags;
         }
     }
-    
+
+    /// <summary>
+    /// Required node for a given Graph. 
+    /// this node will be created automatically in the chart when it is created.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = true)]
+    public class RequireNodeAttribute : Attribute
+    {
+        public Type type { get; private set; }
+        public string nodeName { get; private set; }
+        public Vector2 position { get; private set; }
+
+        /// <summary>
+        /// NodeType Required
+        /// </summary>
+        /// <param name="type">Type of the node</param>
+        public RequireNodeAttribute(Type type)
+        {
+            this.type = type;
+
+        }
+        /// <summary>
+        /// NodeType required
+        /// </summary>
+        /// <param name="type">Type of the node</param>
+        /// <param name="nodeName">Header title name of the node at graph</param>
+        public RequireNodeAttribute(Type type, string nodeName = "")
+        {
+            this.type = type;
+            this.nodeName = nodeName;
+
+        }
+        /// <summary>
+        /// NodeType required
+        /// </summary>
+        /// <param name="type">Type of the node</param>
+        /// <param name="nodeName">Header title name of the node at graph</param>
+        /// <param name="randonPosition">If true, the node will be created at random position at graph. Default X: (-100,100) Y:(-100,100)</param>
+        public RequireNodeAttribute(Type type, string nodeName = "", bool randonPosition = false)
+        {
+            this.type = type;
+            this.nodeName = nodeName;
+            if (randonPosition)
+            {
+                position = new Vector2(UnityEngine.Random.Range(-100, 100), UnityEngine.Random.Range(-100, 100));
```

Comment:
Let's leave out the random position feature. I can't see a lot of use cases for it and there's a lot of situations where it may not act desirably (e.g. still overlaps randomly placed nodes).

If someone wants initial placement randomized it seems like a good reason to implement that on their onw in `Graph.OnGraphEnable()`.